src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -158,7 +161,7 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
                     Debug.Assert(sslHostName == null);
                     Debug.Assert(proxyUri != null);
 
-                    _http2Enabled = false;
+                    _http2Enabled = false; // ToDo: why?
```

Comment:
We don't support HTTP/2 proxies, only HTTP/1. This path is for a CONNECT proxy, which is used for HTTPS requests.

Reply:
> We don't support HTTP/2 proxies, only HTTP/1

And what about `SslProxyTunnel`? It's enabled for H2.

>This path is for a CONNECT proxy, which is used for HTTPS requests.

I thought that's `SslProxyTunnel`.

**How I understand it, now:**
`HttpConnectionKind.Proxy` -> H1 only, ordinary proxy that interprets requests and send them on the client's behalf.
`HttpConnectionKind.ProxyConnect` -> H1 only, special pool just for sending `CONNECT` request to create a tunnel.
`HttpConnectionKind.ProxyTunnel` -> H1 only, web socket, created by `CONNECT` request.
`HttpConnectionKind.SslProxyTunnel` -> H1 or H2, TLS tunnelling through proxy, created by `CONNECT` request.